api/types.go

```diff
@@ -609,6 +617,48 @@ func (pins PinSerial) ToPin() Pin {
 	}
 }
 
+// NodeSerial encodes an ipld node.  The cid and raw data are bundled together
+// to be decoded into a block using NewBlockWithCid which can further be
+// decoded into an ipld node
+type NodeSerial struct {
+	CidS string
+	Data []byte
+}
+
+// ToIPLDNode converts a node serial to an ipld DAG node.  Data must be
+// encoded as Raw or DagCBOR (v0 or v1) for this to work
+func (nS NodeSerial) ToIPLDNode() (ipld.Node, error) {
```

Comment:
Hmm, I did notice this and put off thinking about breaking the pattern.  In light of your observation that we do not need the node itself, just the data and metadata, returning an error is much more sensible to avoid (ToIPLDNode will go away). Moving to this approach I am happy keeping the old error-less policy here for now.

Also happy to discuss changing the policy however.  I don't feel too strongly either way.